homeassistant/components/stream/__init__.py

```diff
@@ -136,7 +136,7 @@ def add_provider(self, fmt, timeout=OUTPUT_IDLE_TIMEOUT):
 
             @callback
             def idle_callback():
-                if not self.keepalive and fmt in self._outputs:
+                if (not self.keepalive or fmt == "recorder") and fmt in self._outputs:
```

Comment:
What is the intent of this change? I thought this was still needed to run even for recorder.

My impression is that the intent here is that "idle" callback is effectively used to end recording of the stream after duration.  That is, we have one callback per format.  The recorder never is "viewed" therefore it never keeps the idle timer awake.

Are there other bugs here?

Clarifying how this worked was my favorite part of PR #46610  I could consider ways to bring this part of the change back.

Reply:
I see: keepalive is preventing this provider from being removed, and therefore you can only have one recording per stream ever? I think i'll add some tests to exercise this in a separate PR.

Reply:
Yes, keepalive was preventing the recorder output from getting removed and cleaning up (running the `recorder_save_worker`)